File: C_Programming/Study/Q5_Assignment15.c

///////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  DisplayRange
// Output:          Search element in the range.
// Description:     Assignment_15: Q.5  : Accept N Number from user & return product of all Odd elements.
//                                        
// Date:            05/06/2026
// Author:          Shital Ajit Nikam   
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////
#include <stdio.h>
#include <stdlib.h>


// time complexity : o(N)
int Product(int Arr[], int iSize)
{
    int iCnt = 0;
    int iProd = 1;

    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        if(Arr[iCnt] % 2 != 0)
        {       
           iProd = iProd * Arr[iCnt];
        }                       
    }
    
    return iProd;
}

int main()
{
    int *Brr = NULL;
    int iLength = 0 , iCnt = 0, iRet = 0;
  
    printf("Enter the number of elements \n");
    scanf("%d",&iLength);
    
    Brr = (int *)malloc(iLength * sizeof(int));

    if(Brr == NULL)
    {
        printf("Unable to allocate Memory");
        return -1;
    }

    printf("Enter %d the Elements \n", iLength);

    for(iCnt = 0; iCnt < iLength ; iCnt++)
    {
        printf("Enter Element: %d", iCnt+1);
        scanf("%d",&Brr[iCnt]);
    }

    
    iRet = Product(Brr, iLength);

    printf("Product of Odd elements is: %d",iRet);
      
    free(Brr);

    return 0;
}


/////////////////////////////////////////////////////////////////////////////////////
//Time Complexity : O(N)  where N = Number of elements
/////////////////////////////////////////////////////////////////////////////////////
// Input : N : 6
//        Elements : 15   66   3   10  44  70
//         
// Output : 45
// 
// 
/////////////////////////////////////////////////////////////////////////////////////

File: Python_Programming/Competative/Q5_Assignment15.py
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name :  Maximum      
# Description:     Assignment_15: Q.5: Write a lambda function using reduce() which accepts list of numbers & returns 
#                                      Maximum element.
# Input  :         [12,33,4,25,9]
# Output :         33                            
#                                     
# Date:            2/07/2026
# Author:          Shital Ajit Nikam   
#
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

from functools import reduce

Maximum = lambda No1,No2: No1 if(No1 > No2) else No2

def main():
    Data = [12,33,4,25,9]
    print("Input Data is :",Data)     
    
    RData = reduce(Maximum, Data)              # Reduce internally calls Maximum & apply it on Data
    print("Maximum number is: ",RData)    
    
if __name__ == "__main__":
    main()